sonic-ledd/scripts/ledd

```diff
@@ -204,9 +206,10 @@ def main():
 
     # Listen indefinitely for changes to the PORT table in the Application DB
     while True:
-        (state, c) = sel.select()
+        (state, c) = sel.select(SELECT_TIMEOUT)
         if state != swsscommon.Select.OBJECT:
-            log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+            if state != swsscommon.Select.TIMEOUT:
+                log_warning("sel.select() did not return swsscommon.Select.OBJECT")
```

Comment:
My main concern here is that it's not clear from the code why you needed to introduce timeout here?
What was the reason to have timeout. I think it's better to have a comment explaining that we introduced the timeout because of the select blocking UNIX signals.

Reply:
Then why C++ code that uses timeout does not tell that in comment?

Is it not enough commit message description? Anyone working on code can use standard approach (e.g. git-log(1), git-blame(1), git-describe(1), etc) to find commit and detailed description of the problem.

If we really want to add comment it might be better to do near function call, because checking return serves completely different purpose: skip unwanted logging messages.

Reply:
I think a comment above the `select()` call on line 209 mentioning that we call select() with a timeout value to to prevent indefinite blocking an enable graceful shutdown via SIGTERM.

Reply:
I checked sonic-swss repo and I found that all select TIMEOUTs there are used to do some actions.
In the code you introduces it's not clear why do we need TIMEOUTs. I would not by surprised if someone would remove the TIMEOUT code as redundant.
I'm sorry I put my comment on the wrong line in the code.

Reply:
@jleveque yes, I agree.

Reply:
Accepted. Thanks.